lib/dul_hydra/batch/models/batch_object_attribute.rb

```diff
@@ -7,9 +7,41 @@ class BatchObjectAttribute < ActiveRecord::Base
     OPERATION_DELETE = "DELETE"    # Delete the provided value from the attribute
     OPERATION_CLEAR = "CLEAR"      # Clear all values from the attribute
 
-    VALUE_TYPE_STRING = "string"
+    OPERATIONS = [ OPERATION_ADD, OPERATION_DELETE, OPERATION_CLEAR ]
 
-    VALUE_TYPES = [ VALUE_TYPE_STRING ]  
-  end
+    VALUE_TYPE_STRING = "STRING"
+
+    VALUE_TYPES = [ VALUE_TYPE_STRING ]
+
+    validates :operation, inclusion: { in: OPERATIONS }
+    validates :datastream, presence: true
+    validates :name, presence: true
+    validate :valid_datastream_and_attribute_name, if: [ 'batch_object.model', 'datastream', 'name' ]
+    with_options if: :operation_requires_value? do |obj|
+      obj.validates :value, presence: true
+      obj.validates :value_type, inclusion: { in: VALUE_TYPES }
+    end
+
+    def operation_requires_value?
+      [ OPERATION_ADD, OPERATION_DELETE ].include? operation
+    end
 
+    def valid_datastream_and_attribute_name
+      if datastream_valid?
+        errors.add(:name, "is not valid") unless attribute_name_valid?
+      else
+        errors.add(:datastream, "is not valid")
+      end
+    end
+
+    def datastream_valid?
+      batch_object.model.constantize.ds_specs[datastream] &&
+        batch_object.model.constantize.ds_specs[datastream][:type] < ActiveFedora::RDFDatastream
+    end
```

Comment:
I would suggest adding a method such as

```ruby
def datastream_type
  batch_object.model.constantize.ds_specs[datastream][:type]
end
```

You could simply allow this method to raise an exception, or add `rescue nil`.

Then, in `datastream_valid?` you simply have:

```ruby
datastream_type < ActiveFedora::RDFDatastream
```